src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -25,6 +25,7 @@ public partial class Frame : FrameContext, IFrameControl
         private static readonly ArraySegment<byte> _continueBytes = CreateAsciiByteArraySegment("HTTP/1.1 100 Continue\r\n\r\n");
         private static readonly ArraySegment<byte> _emptyData = new ArraySegment<byte>(new byte[0]);
         private static readonly byte[] _hex = Encoding.ASCII.GetBytes("0123456789abcdef");
+        private static readonly Stream _nullStream = new NullStream();
```

Comment:
We could use Dispose pattern on MessageBody to consume data that application not read to move input stream instead of read to buffer and write null. I can provide pull request.